adbc.h

```diff
@@ -459,6 +459,28 @@ const struct AdbcError* AdbcErrorFromArrayStream(struct ArrowArrayStream* stream
 ///
 /// \see AdbcConnectionGetInfo
 #define ADBC_INFO_VENDOR_ARROW_VERSION 2
+/// \brief Indicates whether the driver connection is read only (type: bool).
+///
+/// \see AdbcConnectionGetInfo
+#define ADBC_INFO_VENDOR_READ_ONLY 3
```

Comment:
Hmm. Should this be an actual option (GetOption/SetOption), since it in principle is mutable in other APIs? e.g. [JDBC `Connection#setReadOnly`](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#setReadOnly-boolean-). Flight SQL of course doesn't let you set this, but it may apply to other things in the future.

Reply:
Good point. I actually don't need this particular option, it just seemed useful while I was adding the others. So perhaps it's a better idea to leave it out (and maybe add it to options.h). I can update this tomorrow.

Reply:
I just pushed this change. It turns out that adbc.h already has `ADBC_CONNECTION_OPTION_READ_ONLY` defined, so I didn't need to add anything to options.h.

Thanks for the review, it would've been a mistake to make this an info code.